File: Client/src/models/screen.ts
export class Screen {
  /** Returns the identifier. */
  public DevicePath: string = "";

  /** Returns the friendly name. */
  public Label: string = "";

  /** Returns whether this screen Is the primary screen. */
  public IsPrimary: boolean = false;

  /** Returns the X position. */
  public X: number = 0;

  /** Returns the Y position. */
  public Y: number = 0;

  /** Returns the width. */
  public Width: number = 0;

  /** Returns the height. */
  public Height: number = 0;

  /** Returns the width in percentages. */
  public WidthPercentage: number = 0;

  /** Returns the height in percentages. */
  public HeightPercentage: number = 0;

  /** Returns the X position in percentages. */
  public XPercentage: number = 0;

  /** Returns the Y position in percentages. */
  public YPercentage: number = 0;

  /** Returns if the screen is selected. */
  public IsSelected: boolean = false;

  public constructor(model?: Partial<Screen>) {
    Object.assign(this, model);
  }
}

File: Server/Entities/Screen.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.InteropServices;

namespace ScreenTemperature.Entities;

[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Auto)]
public struct PHYSICAL_MONITOR
{
    public IntPtr hPhysicalMonitor;

    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 128)]
    public string szPhysicalMonitorDescription;
}


/// <summary>
/// Represents a screen attached to this machine.
/// </summary>
[NotMapped]
public class Screen
{
    /// <summary>
    /// Returns the identifier.
    /// </summary>
    public string DevicePath { get; set; }

    /// <summary>
    /// Returns the friendly name.
    /// </summary>
    public string Label { get; set; }

    /// <summary>
    /// Returns whether this screen Is the primary screen.
    /// </summary>
    public bool IsPrimary { get; set; }
    
    /// <summary>
    /// Returns the X position.
    /// </summary>
    public int X { get; set; }

    /// <summary>
    /// Returns the Y position.
    /// </summary>
    public int Y { get; set; }

    /// <summary>
    /// Returns the width.
    /// </summary>
    public int Width { get; set; }

    /// <summary>
    /// Returns the height.
    /// </summary>
    public int Height { get; set; }

    public bool IsDDCSupported { get; set; }

    public bool IsBrightnessSupported { get; set; }

    public uint MaxBrightness { get; set; }
    public uint MinBrightness { get; set; }
    public uint CurrentBrightness { get; set; }

    public PHYSICAL_MONITOR[] PhysicalMonitors { get; set; }
}